### PR TITLE
Bounty and amount_to_be_burned calculated directly on stake and denominator for penalty calculation made common across

### LINKED ISSUE #460: combine denominator parameters such as penalty denominators

### PR DESCRIPTION
**Pull request by abhishekvispute**
Fixes #460 
And following slither warning

```
StakeManager.slash(uint32,uint32,address) (contracts/Core/StakeManager.sol#289-316) performs a multiplication on the result of a division:
	-slashPenaltyAmount = (_stake * parameters.slashPenaltyNum()) / parameters.slashPenaltyDenom() (contracts/Core/StakeManager.sol#295)
	-bounty = (slashPenaltyAmount * parameters.bountyNum()) / parameters.bountyDenom() (contracts/Core/StakeManager.sol#298)
Reference: https://github.com/crytic/slither/wiki/Detector-Documentation#divide-before-multiply
. analyzed (66 contracts with 73 detectors), 1 result(s) found
```

### CODE REVIEW

**Comment by abhishekvispute** [comment]
and slither is passed :)

File: contracts/Core/Parameters.sol [hunk 1]
```diff
@@ -37,39 +38,21 @@ contract Parameters is ACL, Constants, IParameters {
...
-
-    function setBurnSlashNum(uint16 _burnSlashNum) external onlyRole(DEFAULT_ADMIN_ROLE) {
-        emit ParameterChanged(msg.sender, "burnSlashNum", burnSlashNum, _burnSlashNum, block.timestamp);
-        burnSlashNum = _burnSlashNum;
+    function setSlashParams(
+        uint16 _bounty,
+        uint16 _burn,
+        uint16 _keep
+    ) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        require(_bounty + _burn + _keep <= 10000, "Slash nums addtion exceeds 10000");
```
Comment (hrishikeshio) [inline contracts/Core/Parameters.sol:46]: 10000 should be replaced by baseDenominator

File: contracts/Core/StakeManager.sol [hunk 2]
```diff
@@ -292,20 +292,28 @@ contract StakeManager is Initializable, ACL, StakeStorage, StateManager, Pause,
...
-        _stake = _stake - slashPenaltyAmount;
-        // prettier ignore
-        uint256 bounty = (slashPenaltyAmount * parameters.bountyNum()) / parameters.bountyDenom();
 
-        if (bounty == 0) return 0;
+        uint256 bounty;
+        uint256 amountToBeBurned;
+        uint256 amountToBeKept;
+
+        // Done for stack too deep issue
```
Comment (SkandaBhat) [inline contracts/Core/StakeManager.sol:300]: Pls describe the issue a bit more

File: contracts/Core/Parameters.sol (hunk 1 above)
Comment (abhishekvispute) [reply contracts/Core/Parameters.sol:46]: sure

File: contracts/Core/StakeManager.sol (hunk 2 above)
Comment (abhishekvispute) [reply contracts/Core/StakeManager.sol:300]: sure : https://medium.com/1milliondevs/compilererror-stack-too-deep-try-removing-local-variables-solved-a6bcecc16231

Comment (abhishekvispute) [reply contracts/Core/StakeManager.sol:300]: basically there is limit upto which stack can store variables
its I think 16 at a moment 
so by creating struct, its stored in memory 
and its ref is in stack 
so it consumes only 1 place

Comment (SkandaBhat) [reply contracts/Core/StakeManager.sol:300]: I meant in the comment, for easy reference later on

Comment (abhishekvispute) [reply contracts/Core/StakeManager.sol:300]: oh okay

Comment (abhishekvispute) [reply contracts/Core/StakeManager.sol:301]: done https://github.com/razor-network/contracts/pull/471/commits/f7f32af3fa82f9dafca4e3750054cf79453d2d8b

File: contracts/Core/Parameters.sol (hunk 1 above)
Comment (abhishekvispute) [reply contracts/Core/Parameters.sol]: done https://github.com/razor-network/contracts/pull/471/commits/f7f32af3fa82f9dafca4e3750054cf79453d2d8b

**Review by SamAg19 — APPROVED** [review]

**Review by SkandaBhat — APPROVED** [review]

**Pull request merged**